antelope_contracts/tests/erc20/integrated_tests.cpp

```diff
@@ -0,0 +1,239 @@
+
+#include <contracts.hpp>
+#include <cstdint>
+#include <cstring>
+#include <eosio/chain/abi_serializer.hpp>
+#include <eosio/chain/fixed_bytes.hpp>
+#include <eosio/testing/tester.hpp>
+#include <fc/crypto/hex.hpp>
+#include <fc/crypto/rand.hpp>
+#include <fc/variant_object.hpp>
+#include <intx/intx.hpp>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include "erc20_tester.hpp"
+
+using namespace eosio;
+using namespace eosio::chain;
+using namespace erc20_test;
+using namespace eosio::testing;
+using mvo = fc::mutable_variant_object;
+
+using intx::operator""_u256;
+constexpr size_t kAddressLength{20};
+
+struct it_tester : erc20_tester {
+    std::string address_str32(const evmc::address& x) {
```

Comment:
so is the resulting string going to be coverted back into binary data and be inserted into call_data? why not just do it by copying the correct position of memory instead of using string operation? for example:
```
    call_data.insert(call_data.end(), 32 - kAddressLength, 0);  // padding for address
    call_data.insert(call_data.end(), address_bytes->begin(), address_bytes->end());
```

Reply:
yes, those code are just a trial (copied from matias's bridge test code) to unify the process to build the calls. 
We may need to find some better way. But the readability should be more important than performance here. I will need to think about how to improve those code.

Reply:
after think about Matias's code, i find it quite hard to write code that is more efficient but keep the same readability. Let's just keep it as it for now.